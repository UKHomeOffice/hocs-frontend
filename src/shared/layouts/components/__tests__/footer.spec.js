import React from 'react';
import Footer from '../footer.jsx';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('Layout footer component', () => {
    it('should render with default props', () => {
        expect(
            render(<Footer />)
        ).toMatchSnapshot();
    });
    it('should render with links when passed', () => {
        const props = {
            links: [
                { label: 'first', target: '/' },
                { label: 'second', target: '/' }
            ]
        };
        expect(
            render(<Footer {...props} />)
        ).toMatchSnapshot();
    });
    it('should show OGL when asked', () => {
        const props = {
            showOGL: true
        };
        expect(
            render(<Footer {...props} />)
        ).toMatchSnapshot();
    });
});
