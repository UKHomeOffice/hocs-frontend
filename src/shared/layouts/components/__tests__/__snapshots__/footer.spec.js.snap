// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout footer component should render with default props 1`] = `
<footer
  class="group js-footer"
  id="footer"
  role="contentinfo"
>
  <div
    class="footer-wrapper"
  >
    <div
      class="footer-meta"
    >
      <div
        class="footer-meta-inner"
      >
        0
        <div
          class="open-government-licence"
        >
          <p
            class="logo"
          >
            <a
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence
            </a>
          </p>
          <p>
            All content is available under the 
            <a
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </p>
        </div>
      </div>
      <div
        class="copyright"
      >
        <a
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/copyright-and-re-use/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`Layout footer component should render with links when passed 1`] = `
<footer
  class="group js-footer"
  id="footer"
  role="contentinfo"
>
  <div
    class="footer-wrapper"
  >
    <div
      class="footer-meta"
    >
      <div
        class="footer-meta-inner"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              first
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              second
            </a>
          </li>
        </ul>
        <div
          class="open-government-licence"
        >
          <p
            class="logo"
          >
            <a
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence
            </a>
          </p>
          <p>
            All content is available under the 
            <a
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </p>
        </div>
      </div>
      <div
        class="copyright"
      >
        <a
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/copyright-and-re-use/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;
